data/event/src/commonMain/kotlin/com/adammcneilly/pocketleague/data/event/StartGGEventService.kt

```diff
@@ -43,11 +44,37 @@ class StartGGEventService(
     }
 
     override suspend fun getEvent(eventId: String): Result<Event> {
-        TODO("Not yet implemented")
+        val eventQuery = TournamentDetailQuery(
+            id = Optional.present(eventId),
+        )
+
+        val response = apiClient.query(eventQuery).execute()
+
+        val event = response.data?.tournament?.tournamentFragment?.toEvent()
+
+        return if (event != null) {
+            Result.success(event)
+        } else {
+            Result.failure(Throwable("Unable to request event for id: $eventId."))
+        }
     }
 
+    /**
+     * Technically this is duplicating an API call, because [getEvent] makes the same GQL fragment query
+     * behind the scenes, but to stay consistent with existing data layers, we're just fetching the event again.
```

Comment:
It might make more sense to have a designated query for participants in an event. 

The reason for this is because if we include the team list in our tournament fragment, that means that we are fetching the teams for every single tournament in a tournament list query. Which works, because GraphQL, but it still means it has to be resolved on the backend, which means we're bringing in more network traffic and also creating unnecessary latency.